### PR TITLE
Fixed failing tests on oldestdeps

### LINKED ISSUE #1401: oldestdeps is using new numpy
In spite of the details in `tox.ini`, on the CI machines at least the `oldestdeps` is using the most recent numpy. This has presumably been going on for a while but it was brought to light by the removal of the deprecated `numpy.isscalar`.

### PR DESCRIPTION
**Pull request by paulray**
Attempt to fix #1401 

Required updating the oldest version of numpy and pinning the versions of scipy and matplotlib to keep it from automatically updating numpy to the most recent.

### CODE REVIEW

**Comment by paulray** [comment]
The reason the bug was not causing a problem before is that that test is skipped on headless machines (e.g. on the CI machines). Can we re-enable that test on CI?

**Comment by paulray** [comment]
I have now removed all the skips if X11 is not available in `test_photonphase.py`. It seems to be passing fine, so I'm not sure why those skips were there. They allowed a bug to persist in the code.

File: src/pint/scripts/photonphase.py [hunk 1]
```diff
@@ -227,7 +227,7 @@ def main(argv=None):
 
         # Calculate phases
         log.debug("Evaluating polycos")
-        phases = p.eval_phase(mjds)
+        phases = ptable.eval_phase(mjds)
```
Comment (paulray) [inline src/pint/scripts/photonphase.py:230]: @sashalvna can you look at this fix? Without this edit, it was crashing for me and the test was not passing. Do you agree it is correct now?

Comment (sashalvna) [reply src/pint/scripts/photonphase.py:230]: Yes, that is correct! ptable contains the polycos that it should be using to calculate the phases.

Comment (paulray) [reply src/pint/scripts/photonphase.py:230]: Thanks much! I can't figure out how it was working before, but it works now!

**Comment by paulray** [comment]
Bizarre, the macos-latest tests ran super slow and got killed after 6 hours. What is going on?

**Comment by paulray** [comment]
OK, this is ready to merge! It solves the CI failures we are getting and increases the test coverage in our CI runs, which revealed a bug.

**Pull request merged**